free-3.md

```diff
@@ -10,41 +10,100 @@ _What need is met by your idea? Who are you helping? What is that person's
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+I want the user's experience to be one of discovery. There is an unimaginably large space to be explored here, and the user should be in charge of how they
+choose to explore that space. The best way to accomplish this seems to be to give the user a small set of building blocks that hopefully form a basis for the space, and to provide an easy way to combine the building blocks to form connections and set off in new directions. In other words, to give the user a language and let them decide what to say with it.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+One of my best friends is an engineering major interested in art, or an artist who is studying engineering. One of the things she is most passionate about is closing the gap between science and art, as I mentioned above. That, and the specific method of exploring math through art or vice versa, are ideas she has often expressed an interest in. As I've learned more about DSLs in this class, it occurred to me that a DSL could be a good way to implement these ideas.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Interactive art described by math.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+Given that this is meant to be an interactive exploratory process, the
+interface should facilitate fast, easy changes to parameters. I imagine it
+would look similar to Bret Victor's
+[system](https://www.youtube.com/watch?v=PUv66718DII&feature=youtu.be&t=571),
+which we saw on the first day of class. The user would be able to define a
+mathematical model for their art -- perhaps using MATLAB-like syntax -- and
+then they could use sliders and other interactive tools to tweak the
+parameters, thereby exploring the space around their model.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+At a very high level, when a program runs, the user's mathematical expressions
+are evaluated and the result is displayed in a visual form. The exact method of
+translating formulas to an image, I have yet to determine. I can think of
+several possibilities. The user could give formulas the determine the color as
+a function of space.
+
+Or, the formulas could be interpreted as describing a set of objects or
+surfaces in 3-space, which are then rendered to the 2D screen.
+
+Or, interesting properties of the functions specified the user could map to
+interesting features in the resulting image. For example, the zeros of a
+function could correspond to bright points or lines in the image.
+
+Or, these methods could be combined in a way specified by the user.
```

Comment:
I think all these methods would be interesting. It might be too much to have everything, but we can see later.